pkg/broker/ensure.go

```diff
@@ -0,0 +1,66 @@
+package broker
+
+import (
+	"fmt"
+
+	apiextension "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+func Ensure(config *rest.Config, ipsecPSKBytes int) error {
+	// Create the CRDs we need
+	apiext, err := apiextension.NewForConfig(config)
+	if err != nil {
+		return fmt.Errorf("error creating the api extensions client: %s", err)
+	}
+	_, err = apiext.ApiextensionsV1beta1().CustomResourceDefinitions().Create(NewClustersCRD())
+	if err != nil && !apierrors.IsAlreadyExists(err) {
+		return fmt.Errorf("error creating the Cluster CRD: %s", err)
+	}
+	_, err = apiext.ApiextensionsV1beta1().CustomResourceDefinitions().Create(NewEndpointsCRD())
+	if err != nil && !apierrors.IsAlreadyExists(err) {
+		return fmt.Errorf("error creating the Endpoint CRD: %s", err)
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return fmt.Errorf("error creating the core kubernetes clientset: %s", err)
+	}
+	// Create the namespace
```

Comment:
Bit of a nitpick, but this comment and a few of the following give the impression they’re attached to the code preceding them, rather than the code following them; a blank line before, and none after, would be clearer.

Reply:
nitpick accepted :)